addons/point_of_sale/static/src/js/Screens/ProductScreen/ProductsWidgetControlPanel.js

```diff
@@ -19,7 +19,24 @@ export class ProductsWidgetControlPanel extends Component {
         this.notification = useService("pos_notification");
         this.orm = useService("orm");
         this.updateSearch = debounce(this.updateSearch, 100);
-        this.state = useState({ mobileSearchBarIsShown: false });
+        this.state = useState({ mobileSearchBarIsShown: false, isMobile: false });
+        this.productsWidgetControl = useRef("products-widget-control");
+        this.ui = useService("ui");
+
+        onMounted(() => {
+            this.toggleIsMobile();
+            window.addEventListener("resize", this.toggleIsMobile.bind(this));
+        });
+        onWillUnmount(() => {
+            window.removeEventListener("resize", this.toggleIsMobile.bind(this));
+        });
```

Comment:
```suggestion
        const _toggleIsMobile = () => this.toggleIsMobile();
        onMounted(() => {
            _toggleIsMobile();
            window.addEventListener("resize", _toggleIsMobile);
        });
        onWillUnmount(() => {
            window.removeEventListener("resize", _toggleIsMobile);
        });
```

Calling `.bind` twice returns different functions. So this way of removing the listener won't work. Check the following screenshot.

<img width="264" alt="Screenshot 2023-05-23 at 10 06 03" src="https://github.com/odoo/odoo/assets/3245568/9eb72268-d1d8-41d3-b498-d70f0d1598fe">

In the proposed change, `_toggleIsMobile` is the `bound` function for `this.toggleIsMobile`.

If you want to get rid of this binding problem, you can define the `toggleIsMobile` locally in the `setup` like so:

```js
    const toggleIsMobile = () => {
        // In addition to the UI service we need to check the width of the window
        // because the search bar glitches on a specific width between when the
        // UI is considered small and not.
        const element = this.productsWidgetControl.el;
        this.state.isMobile = (element && element.offsetWidth < 350) || this.ui.isSmall;
    }
onMounted(() => {
            toggleIsMobile();
            window.addEventListener("resize", toggleIsMobile);
        });
        onWillUnmount(() => {
            window.removeEventListener("resize", toggleIsMobile);
        });
```

Defining the callback locally will prevent other devs from overriding it.